In [12]:
import pandas as pd

#load dataset 
df = pd.read_csv("Sales_Dataset.csv")
df.head()

,Order_ID,Order_Date,Customer,City,Region,Product,Category,Sales,Quantity,Discount,Profit
0,1001,2024-10-01,Priya,Hyderabad,East,Mouse,Accessories,1735,7,0.00,307.52
1,1002,2024-05-03,Sneha,Kolkata,East,Monitor,Electronics,807,2,0.15,49.89
2,1003,2024-01-26,Anita,Mumbai,West,Phone,Electronics,2328,1,0.20,66.47
3,1004,2024-06-06,Sara,Pune,East,Keyboard,Accessories,3933,3,0.20,-71.66
4,1005,2024-10-15,Rahul,Bangalore,South,Monitor,Electronics,3550,2,0.20,146.30


#**Q1 Dataset Exploration
Perform the following exploratory steps:**
Display the first 10 rows;

Display the dataset structure using info();

Find the number of rows and columns;

Identify the data types of each column

In [13]:
#**solution**

#Display the first 10 rows
print("First 10 rows:")
print(df.head(10))


#Display the dataset structure using info();
print("\nInfo of dataset df:") 
print(df.info())

#Find the number of rows and columns;
print("\nDataset Shape:")
print(df.shape)


#Identify the data types of each column
print("\nData Types:")
df.dtypes

First 10 rows:
   Order_ID  Order_Date Customer       City Region   Product     Category  \
0      1001  2024-10-01    Priya  Hyderabad   East     Mouse  Accessories   
1      1002  2024-05-03    Sneha    Kolkata   East   Monitor  Electronics   
2      1003  2024-01-26    Anita     Mumbai   West     Phone  Electronics   
3      1004  2024-06-06     Sara       Pune   East  Keyboard  Accessories   
4      1005  2024-10-15    Rahul  Bangalore  South   Monitor  Electronics   
5      1006  2024-07-04    Pooja  Bangalore   West   Printer       Office   
6      1007  2024-06-24      Dev    Chennai   East   Monitor  Electronics   
7      1008  2024-03-18    Kavya    Kolkata   East     Phone  Electronics   
8      1009  2024-10-31    Kavya  Ahmedabad   East     Phone  Electronics   
9      1010  2024-12-25    Pooja  Ahmedabad  North     Phone  Electronics   

   Sales  Quantity  Discount  Profit  
0   1735         7      0.00  307.52  
1    807         2      0.15   49.89  
2   2328         1  

Order_ID        int64
Order_Date     object
Customer       object
City           object
Region         object
Product        object
Category       object
Sales           int64
Quantity        int64
Discount      float64
Profit        float64
dtype: object

#**Q2 Sales Filtering Analysis**
Filter the dataset to display:
Orders where Sales > 3500 ;

Orders where Region = "East" and Sales > 2500 ;

Explain what insights you observe

In [14]:
#1. Orders where Sales > 3500
high_sales = df[df['Sales'] > 3500]

print("Orders with Sales > 3500:")
print(high_sales.head())


# 2. Orders where Region = "East" and Sales > 2500
east_high_sales = df[(df['Region'] == 'East') & (df['Sales'] > 2500)]

print("\nEast Region Orders with Sales > 2500:")
print(east_high_sales.head())

Orders with Sales > 3500:
    Order_ID  Order_Date Customer       City Region   Product     Category  \
3       1004  2024-06-06     Sara       Pune   East  Keyboard  Accessories   
4       1005  2024-10-15    Rahul  Bangalore  South   Monitor  Electronics   
7       1008  2024-03-18    Kavya    Kolkata   East     Phone  Electronics   
8       1009  2024-10-31    Kavya  Ahmedabad   East     Phone  Electronics   
10      1011  2024-06-30     Neha      Delhi  South   Printer       Office   

    Sales  Quantity  Discount   Profit  
3    3933         3      0.20   -71.66  
4    3550         2      0.20   146.30  
7    4693         7      0.05   918.49  
8    5071         6      0.10   707.55  
10   5977         6      0.00  1578.32  

East Region Orders with Sales > 2500:
    Order_ID  Order_Date Customer       City Region   Product     Category  \
3       1004  2024-06-06     Sara       Pune   East  Keyboard  Accessories   
7       1008  2024-03-18    Kavya    Kolkata   East     Phone  E

#*Insights*

    - Orders with sales greater than 3500 represent high-value transactions, indicating premium customers or bulk purchases.
    
    -Filtering East region with sales above 2500 shows region-specific strong sales performance.
    
    -If many rows appear in East, it suggests East region has better sales potential or higher demand.
    
    -This type of filtering helps businesses target high-performing regions and customers.


**Q3 Customer Purchase Analysis**

    Identify the top 5 customers based on total sales.
Hint: Use GroupBy with aggregation

In [18]:
customer_sales = df.groupby('Customer')['Sales'].sum()        # Group by Customer and calculate total sales

top_customers = customer_sales.sort_values(ascending=False)      # Sort in descending order

top_5_customers = top_customers.head(5)          # Get top 5 customers

print("Top 5 Customers based on Total Sales:")
print(top_5_customers)

Top 5 Customers based on Total Sales:
Customer
Nitin    59820
Raj      58416
Rahul    52291
Dev      48500
Isha     46818
Name: Sales, dtype: int64


*insights:*
    Top 5 customers contribute a significant portion of total revenue, indicating customer concentration risk or opportunity.

**Q4 Regional Sales Performance.
Calculate the total sales generated by each region.**

Then identify:
Which region generated the highest revenue

In [27]:
# Total sales by region
region_sales = df.groupby('Region')['Sales'].sum()

print("Total Sales by Region:")
print(region_sales)

#Region with the highest revenue
print("\nRegion with the highest revenue :")
print( region_sales.head(1))

Total Sales by Region:
Region
East     232675
North    158210
South    183108
West     170227
Name: Sales, dtype: int64

Region with the highest revenue :
Region
East    232675
Name: Sales, dtype: int64


**Interpretation:** 

    Sales were grouped by region to evaluate regional performance.
    
    The region with the highest total sales is"East". It represents the strongest market

**Q5 Product Performance Analysis**

Calculate the total quantity sold for each product.
Which product appears to be the most popular?

In [32]:
top_product = df.groupby('Product')['Quantity'].sum().nlargest(1)

print("Top Product:\n" , 
        top_product)

Top Product:
 Product
Monitor    156
Name: Quantity, dtype: int64


**Insights**

    The dataset was grouped by product to calculate total quantity sold.
    
    The product with the highest quantity sold is "Monitor" & considered the most popular product

**Q6 Multi-Level Grouping**

Group the dataset by: Region and Category. 
Calculate the total sales for each combination.
What insights can you derive from this analysis?

In [34]:
# Multi-level grouping
multi_group = df.groupby(['Region', 'Category'])['Sales'].sum().reset_index()

print("Total Sales by Region and Category:\n")
print(multi_group)

Total Sales by Region and Category:

   Region     Category   Sales
0    East  Accessories   57646
1    East  Electronics  160907
2    East       Office   14122
3   North  Accessories   62052
4   North  Electronics   70023
5   North       Office   26135
6   South  Accessories   80914
7   South  Electronics   86690
8   South       Office   15504
9    West  Accessories   64811
10   West  Electronics   84128
11   West       Office   21288


In [35]:
multi_group = multi_group.sort_values(by='Sales', ascending=False)

print("\nSorted Sales (High to Low):\n")
print(multi_group)


Sorted Sales (High to Low):

   Region     Category   Sales
1    East  Electronics  160907
7   South  Electronics   86690
10   West  Electronics   84128
6   South  Accessories   80914
4   North  Electronics   70023
9    West  Accessories   64811
3   North  Accessories   62052
0    East  Accessories   57646
5   North       Office   26135
11   West       Office   21288
8   South       Office   15504
2    East       Office   14122


**INSIGHTS**
  
    -Electronics category performs highest in the East region.
    
    -Office sales are lower in some regions, indicating weaker demand.
    
    -Electronics Supplies show consistent performance across regions.

**Q7 Pivot Table Creation**
Create a pivot table showing:

Rows → Region

Columns → Product

Values → Total Sales

Which region sells the most laptops?

In [39]:
# Create pivot table
pivot_table = df.pivot_table(
    values='Sales',
    index='Region',
    columns='Product',
    aggfunc='sum'
)

print("Pivot Table:\n")
print(pivot_table)

Pivot Table:

Product  Headphones  Keyboard  Laptop  Monitor  Mouse  Phone  Printer  Tablet
Region                                                                       
East          23899     10845   55568    31299  22902  39667    14122   34373
North         15450     20538    4540    14922  26064  30535    26135   20026
South         24111     23480   23029    19167  33323  18218    15504   26276
West          17159     38499   10826    18178   9153  27850    21288   27274


In [40]:
#region with highest Laptop sales

print("\nLaptop Sales by Region:\n")
print(pivot_table['Laptop'].sort_values(ascending=False))


Laptop Sales by Region:

Region
East     55568
South    23029
West     10826
North     4540
Name: Laptop, dtype: int64


**INSIGHTS**

    The EAST region has the highest laptop sales, indicating strong demand.
    
    Other regions show comparatively lower sales, suggesting growth opportunities.

**Q8 Profitability Analysis**
    Calculate the average profit for each product category.
Which category is most profitable?

In [42]:
# Average profit per category
category_profit = df.groupby('Category')['Profit'].mean().reset_index()

# Sort descending
category_profit = category_profit.sort_values(by='Profit', ascending=False)

print("Average Profit by Category:\n")
print(category_profit)

Average Profit by Category:

      Category      Profit
2       Office  471.938462
0  Accessories  410.603086
1  Electronics  396.462114


In [43]:
#Most Profitable Category
top_category = df.groupby('Category')['Profit'].mean().nlargest(1)

print(top_category)

Category
Office    471.938462
Name: Profit, dtype: float64


MOST PROFITABLE CATEGORY IS : **Office**

**Q9 Discount Impact Study**

Analyze the relationship between discount and profit.
Tasks:

Calculate the average profit for each discount level

Identify whether higher discounts lead to lower profit

In [44]:
# Average profit for each discount level
discount_profit = df.groupby('Discount')['Profit'].mean().reset_index()

# Sort by discount
discount_profit = discount_profit.sort_values(by='Discount')

print("Average Profit by Discount Level:\n")
print(discount_profit)

Average Profit by Discount Level:

   Discount      Profit
0      0.00  751.210588
1      0.05  531.368039
2      0.10  346.460513
3      0.15  237.191400
4      0.20   90.026923


**INSIGHTS**

    The average profit decreases consistently as the discount increases.
    
    At 0% discount, profit is highest (~751), indicating maximum margin.

    Even a small discount (5%) reduces profit significantly.

    At 20% discount, profit drops drastically (~90), showing very low profitability.

    This clearly indicates a strong negative relationship between discount and profit.

**Q10 End-to-End Data Wrangling Task**

Perform the following workflow:

Filter the dataset where Sales > 3000

Group data by Region and Product

Calculate total sales

Reshape the result into a pivot table

Explain the business insights obtained from this analysis

In [45]:
high_sales = df[df['Sales'] > 3000]

print("Filtered Data (Sales > 3000):\n")
print(high_sales.head())


Filtered Data (Sales > 3000):

   Order_ID  Order_Date Customer       City Region   Product     Category  \
3      1004  2024-06-06     Sara       Pune   East  Keyboard  Accessories   
4      1005  2024-10-15    Rahul  Bangalore  South   Monitor  Electronics   
5      1006  2024-07-04    Pooja  Bangalore   West   Printer       Office   
7      1008  2024-03-18    Kavya    Kolkata   East     Phone  Electronics   
8      1009  2024-10-31    Kavya  Ahmedabad   East     Phone  Electronics   

   Sales  Quantity  Discount  Profit  
3   3933         3      0.20  -71.66  
4   3550         2      0.20  146.30  
5   3073         8      0.20  191.58  
7   4693         7      0.05  918.49  
8   5071         6      0.10  707.55  


In [46]:
grouped_data = high_sales.groupby(['Region', 'Product'])['Sales'].sum().reset_index()

print("\nGrouped Data:\n")
print(grouped_data)


Grouped Data:

   Region     Product  Sales
0    East  Headphones  14469
1    East    Keyboard   9642
2    East      Laptop  44096
3    East     Monitor  16420
4    East       Mouse  19000
5    East       Phone  29500
6    East     Printer   3696
7    East      Tablet  29229
8   North  Headphones   5326
9   North    Keyboard  15653
10  North      Laptop   4540
11  North     Monitor   5087
12  North       Mouse  22209
13  North       Phone  24670
14  North     Printer  17185
15  North      Tablet   3392
16  South  Headphones  10026
17  South    Keyboard  19249
18  South      Laptop  17950
19  South     Monitor  14921
20  South       Mouse  26928
21  South       Phone  18218
22  South     Printer  14143
23  South      Tablet  19613
24   West  Headphones  13898
25   West    Keyboard  33380
26   West      Laptop   5922
27   West     Monitor  14354
28   West       Mouse   3511
29   West       Phone  15653
30   West     Printer  12648
31   West      Tablet  17185


In [47]:
pivot_result = grouped_data.pivot(
    index='Region',
    columns='Product',
    values='Sales'
)

print("\nFinal Pivot Table:\n")
print(pivot_result)


Final Pivot Table:

Product  Headphones  Keyboard  Laptop  Monitor  Mouse  Phone  Printer  Tablet
Region                                                                       
East          14469      9642   44096    16420  19000  29500     3696   29229
North          5326     15653    4540     5087  22209  24670    17185    3392
South         10026     19249   17950    14921  26928  18218    14143   19613
West          13898     33380    5922    14354   3511  15653    12648   17185


**Explanation**

    The dataset was filtered to include only high-value transactions (Sales > 3000).

    The filtered data was grouped by region and product to analyze performance.
    
    A pivot table was created to compare product-wise sales across regions.

    This approach helps identify top-performing products in each region.